lib/qs/event.rb

```diff
@@ -0,0 +1,80 @@
+require 'qs/job'
+
+module Qs
+
+  class Event
+
+    def self.build(channel, name, params, published_at = nil)
+      job_name   = EventJobName.new(channel, name)
+      job_params = {
+        'event_channel' => channel,
+        'event_name'    => name,
+        'event_params'  => params
+      }
+      self.new(Qs::Job.new(job_name, job_params, published_at))
+    end
+
+    attr_reader :job
+
+    def initialize(job)
+      validate!(job)
+      @job = job
+    end
+
+    def channel
+      @job.params['event_channel']
+    end
+
+    def name
+      @job.params['event_name']
+    end
+
+    def params
+      @job.params['event_params']
+    end
+
+    def published_at
+      @job.created_at
+    end
+
+    def inspect
+      reference = '0x0%x' % (self.object_id << 1)
+      "#<#{self.class}:#{reference} " \
+      "@channel=#{self.channel.inspect} " \
+      "@name=#{self.name.inspect} " \
+      "@params=#{self.params.inspect} " \
+      "@published_at=#{self.published_at.inspect}>"
+    end
+
+    def ==(other)
+      if other.kind_of?(self.class)
+        self.job == other.job
+      else
+        super
+      end
+    end
+
+    private
+
+    def validate!(job)
+      problem = if job.params['event_channel'].to_s.empty?
+        "The job doesn't have an event channel."
+      elsif job.params['event_name'].to_s.empty?
+        "The job doesn't have an event name."
+      elsif !job.params['event_params'].kind_of?(::Hash)
+        "The job's event params are not valid."
+      end
+      raise(BadEventError, problem) if problem
+    end
+
+  end
+
+  module EventJobName
+    def self.new(event_channel, event_name)
+      "#{event_channel}_#{event_name}"
```

Comment:
@kellyredding - This will probably expand in some way to support prefixing. I'm not 100% sure how I'm going to handle that with `Job` and it prefixing as well. For now, I'll leave it without any prefixing and address that when I add configuring events on a queue, which is where the issue has to be addressed to avoid collisions between jobs and events.

I'm cool with using `:` to delimit the channel and event.